test/scripts/e2e_subs/e2e-app-cross-round.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+date '+app-stateful-global-test start %Y%m%d_%H%M%S'
+
+set -e
+set -x
+set -o pipefail
+export SHELLOPTS
+
+WALLET=$1
+
+# Directory of this bash program
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+gcmd="goal -w ${WALLET}"
+
+ACCOUNT=$(${gcmd} account list|awk '{ print $3 }')
+
+# Succeed in creating app that approves transactions with arg[0] == 'hello'
+printf '#pragma version 2\nint 1' > "${TEMPDIR}/int1.teal"
+APPID=$(${gcmd} app create --creator ${ACCOUNT} --approval-prog ${DIR}/tealprogs/cross-round.teal --global-byteslices 2 --global-ints 2 --local-byteslices 2 --local-ints 2 --clear-prog "${TEMPDIR}/int1.teal" | grep Created | awk '{ print $6 }')
+
+# Should succeed to opt in with first arg hello
+${gcmd} app optin --app-id $APPID --from $ACCOUNT --app-arg "str:first"
```

Comment:
nit:
just for the sake of readability, could you flip the order of the last two arguments ?
i.e.
```
${gcmd} app optin --app-id $APPID --app-arg "str:first" --from $ACCOUNT
```
instead of
```
${gcmd} app optin --app-id $APPID --from $ACCOUNT --app-arg "str:first"
```